includes/core/classes/class-import.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * Class responsible for importing content using WordPress' native import tool(s).
+ *
+ * @package GatherPress\Core
+ * @since 1.0.0
+ */
+
+namespace GatherPress\Core;
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit; // @codeCoverageIgnore
+
+use GatherPress\Core\Event;
+use GatherPress\Core\Migrate;
+use GatherPress\Core\Traits\Singleton;
+use WP_Post;
+
+/**
+ * Class Import.
+ *
+ * The Import class handles the importing of content using WordPress' native import tool.
+ * This class will provide effective filtering and support validation of the import-objects
+ * based on their post type and meta data.
+ *
+ * Succesfully identified GatherPress data will be saved into custom DB tables.
+ *
+ * @since 1.0.0
+ */
+class Import extends Migrate {
+	/**
+	 * Enforces a single instance of this class.
+	 */
+	use Singleton;
+
+	/**
+	 * Class constructor.
+	 *
+	 * This method initializes the object and sets up necessary hooks.
+	 *
+	 * @since 1.0.0
+	 */
+	public function __construct() {
+		$this->setup_hooks();
+	}
+
+	/**
+	 * Set up hooks for various purposes.
+	 *
+	 * This method adds hooks for different purposes as needed.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	protected function setup_hooks(): void {
+
+		if ( class_exists( 'WXR_Importer' ) ) {
+			/**
+			 * Setup for WordPress Importer (v2).
+			 *
+			 * @see https://github.com/humanmade/Wordpress-Importer
+			 */
+			add_filter( 'wxr_importer.pre_process.post', array( '\GatherPress\Core\Import', 'set_entry_point' ) );
+		} else {
+			/**
+			 * Setup for default WordPress Importer.
+			 *
+			 * @see https://github.com/WordPress/wordpress-importer/issues/42
+			 */
+			add_filter( 'wp_import_post_data_raw', array( '\GatherPress\Core\Import', 'set_entry_point' ) );
+		}
+		add_action( 'gatherpress_import', array( '\GatherPress\Core\Import', 'import' ) );
```

Comment:
I thought I understood what you meant, but went with `self::class`, because I'm referencing only static functions. 
But I'm still not sure, about it. 

Will this work and be ok?

Reply:
Can you please have a Quick Look at this Long open review comment @mauteri ?